crates/fuzzing/src/oracles.rs

```diff
@@ -727,13 +722,135 @@ pub fn differential_wasmi_execution(wasm: &[u8], config: &crate::generators::Con
     }
 
     if &wasmi_buf[..] != &wasmtime_slice[..] {
-        show_wat();
+        show_wat(wasm);
         panic!("memory contents are not equal");
     }
 
     Some(())
 }
 
+/// Perform differential execution between Wasmtime and the official WebAssembly
+/// specification interpreter.
+///
+/// May return `None` if we early-out due to a rejected fuzz config.
+pub fn differential_spec_execution(wasm: &[u8], config: &crate::generators::Config) -> Option<()> {
+    crate::init_fuzzing();
+    debug!("WAT: {}", wasm2wat(wasm));
+    debug!("config: {:#?}", config);
+
+    let wasmtime_vals = run_in_wasmtime(wasm, config, &[]);
+    debug!("Wasmtime returned: {:?}", wasmtime_vals);
+    let spec_vals = wasm_spec_interpreter::interpret(wasm, vec![]);
+    debug!("spec interpreter returned: {:?}", &spec_vals);
+
+    // Match a spec interpreter value against a Wasmtime value. Eventually this
+    // should support references and `v128` (TODO).
+    fn matches(spec_val: &wasm_spec_interpreter::Value, wasmtime_val: &wasmtime::Val) -> bool {
+        match (spec_val, wasmtime_val) {
+            (wasm_spec_interpreter::Value::I32(a), wasmtime::Val::I32(b)) => a == b,
+            (wasm_spec_interpreter::Value::I64(a), wasmtime::Val::I64(b)) => a == b,
+            (wasm_spec_interpreter::Value::F32(a), wasmtime::Val::F32(b)) => *a as u32 == *b,
+            (wasm_spec_interpreter::Value::F64(a), wasmtime::Val::F64(b)) => *a as u64 == *b,
+            (_, _) => false,
```

Comment:
This should probably be `unreachable!()`, since we don't expect to ever get other types of values.